bip-v1-packages.mediawiki

```diff
@@ -0,0 +1,314 @@
+
+# The reciever uses the package information to decide how to request the transactions. For example, if the receiver already has some of the transactions in their mempool, they only request the missing ones. They could also decide not to request the package at all based on the fee information provided.
+
+# Upon receiving a "pckgtxns", the receiver submits the transactions together as a package.
+
+[[File:./bip-v1-packages/normal_child_parents.png|600px]]
+
+''Diagram: The intended protocol flow for a sender relaying zero-fee parents with a high-fee child.''
+
+[[File:./bip-v1-packages/too_low.png|600px]]
+
+''Diagram: A scenario in which a high-fee child is insufficient to pay for its low-fee parents, and no transactions need to be relayed.''
+
+[[File:./bip-v1-packages/cpfp_dedup.png|600px]]
+
+''Diagram: A scenario in which one of the package transactions has already been accepted and is not requested a second time.''
+
+Child-with-unconfirmed-parents package relay is negotiated between two peers through the "sendpackages" message
+with version=1.
+
+[[File:./bip-v1-packages/sendpackages_negotiation.png|600px]]
+
+''Diagram: Version 1 package relay is negotiated during the version handshake.''
+
+===New Messages===
+
+A new inv type, "MSG_PCKG1", and new protocol message, "PCKGINFO1", are added.
+
+====pckginfo1====
+
+{|
+|  Field Name  ||  Type  ||  Size  ||   Purpose
+|-
+|blockhash || uint256 || 32 || The chain tip at which this package is defined.
+|-
+|pckg_fee||CAmount||4|| The sum total fees paid by all transactions in the package.
+|-
+|pckg_weight||int64_t||8|| The sum total weight of all transactions in the package.
+|-
+|txns_length||CompactSize||1 or 3 bytes|| The number of transactions provided.
+|-
+|txns||List of wtxids||txns_length * 32|| The wtxids of each transaction in the package.
+|}
+
+
+# The "pckginfo1" message has the structure defined above, with pchCommand == "pckginfo1".
+
+# A "pckginfo1" message contains information about a version 1 package (defined below), referenced by the wtxid of the transaction it pertains to and the current blockhash.
+
+# Upon receipt of a "pckginfo1" message, the node should decide if it wants to validate the package, request transaction data if necessary, etc.
+
+# Upon receipt of a malformed "pckginfo1" message or package that does not abide by the max_count, max_weight, or other rules specified by the version agreed upon in the initial negotiation, the sender should be disconnected.  If a node receives a "pckginfo1" message for which the "pckg_fee" or "pckg_weight" do not reflect the true total fees and weight, respectively, or the transactions in the package, the message is malformed.
+
+# A node MUST NOT send a "pckginfo1" message that has not been requested by the recipient. Upon receipt of an unsolicited "pckginfo1", a node should disconnect the sender.
```

Comment:
Requested by recipient means a GETDATA(MSG_PCKG1) was sent?  If so I'd suggest making that more explicit.  

Also, I think usually we have left the handling of protocol violations as not themselves mandating particular handling, ie, I'd suggest that the spec be "Upon receipt of an unsolicited pckginfo1, a node MAY disconnect the sender".